bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java

```diff
@@ -213,76 +258,137 @@ public void snapshotAsync(SnapshotDescription snapshot)
   }
 
   @Override
-  public Future<Void> addColumnFamilyAsync(TableName arg0, ColumnFamilyDescriptor arg1)
+  public Future<Void> addColumnFamilyAsync(TableName tableName, ColumnFamilyDescriptor columnFamily)
       throws IOException {
-    throw new UnsupportedOperationException("addColumnFamilyAsync"); 
+    String columnName = columnFamily.getNameAsString();
+    Modification.Builder modification = Modification.newBuilder().setId(columnName)
+        .setCreate(new ColumnDescriptorAdapter()
+        .adapt((HColumnDescriptor) columnFamily).build());
+    return FutureUtils
+        .toCompletableFuture(modifyColumnAsync(tableName, columnName, "add", modification))
+        .thenApply(r -> null);
   }
 
   @Override
-  public List<SnapshotDescription> listSnapshots()
-      throws IOException {
-    throw new IllegalArgumentException("listSnapshots is not supported");
+  public void deleteColumnFamily(TableName tableName, byte[] columnName) throws IOException {
+    deleteColumn(tableName, columnName);
   }
 
-
   @Override
-  public void deleteColumnFamily(TableName arg0, byte[] arg1) throws IOException {
-    throw new UnsupportedOperationException("deleteColumnFamily"); // TODO
+  public Future<Void> deleteColumnFamilyAsync(TableName tableName, byte[] columnName) 
+      throws IOException {
+    final String columnNameStr = Bytes.toString(columnName);
+    Modification.Builder modification = Modification.newBuilder().setId(columnNameStr)
+        .setDrop(true);
+    return FutureUtils.toCompletableFuture(modifyColumnAsync(
+        tableName, 
+        columnNameStr, 
+        "delete", 
+        modification)).thenApply(r -> null);
+  }
+
+
+  protected ListenableFuture<Table> modifyColumnAsync(TableName tableName, String columnName,
+      String modificationType, Modification.Builder modification) throws IOException {
+    ModifyColumnFamiliesRequest.Builder modifyColumnBuilder = ModifyColumnFamiliesRequest
+        .newBuilder().addModifications(modification).setName(toBigtableName(tableName));
+
+    try {
+      return bigtableTableAdminClient.modifyColumnFamilyAsync(modifyColumnBuilder.build());
+    } catch (Throwable throwable) {
+      throw new IOException(
+          String.format(
+              "Failed to %s column '%s' in table '%s'",
+              modificationType,
+              columnName,
+              tableName.getNameAsString()),
+          throwable);
+    }
   }
 
-  @Override
-  public Future<Void> deleteColumnFamilyAsync(TableName arg0, byte[] arg1) throws IOException {
-    throw new UnsupportedOperationException("deleteColumnFamily"); // TODO
+  protected ListenableFuture<Empty> deleteTableAsyncInternal(TableName tableName) 
+      throws IOException {
+    Builder deleteBuilder = DeleteTableRequest.newBuilder();
```

Comment:
Please use `DeleteTableRequest.Builder` instead of `Builder`.  All protobuf objects have a `Builder`, so use of just `Builder gets confusing.  Also, it might be better to just do this:

DeleteTableRequest request = DeleteTableRequest.newBuilder()
    .setName(...)
    .build().

Reply:
Also, please return a `CompletableFuture<Void>` so that the caller doesn't have to.

Reply:
Will do.